h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -463,6 +464,11 @@ public void init(boolean expensive) {
         case gaussian:
 //          if (_nclass != 1) error("_family", H2O.technote(2, "Gaussian requires the response to be numeric."));
           break;
+        case fractionalbinomial:
+          Vec resp = (train()).vec(_parms._response_column);
+          if ((resp.min() <0) || (resp.max()>1))
+            error("response", " response must be between 0 and 1 for fractional_binomial family.");
```

Comment:
```suggestion
          if ((resp.min() < 0) || (resp.max() > 1)) {
            error("response",
                    String.format("Response '%s' must be between 0 and 1 for fractional_binomial family. Min: %f, Max: %f",
                            _parms._response_column, resp.min(), resp.max()));
          }
```

Provide details to the user about which boundary is exceeded. We have this information, why not give it out for free to the user.

Reply:
Also, let's add a test for it into `GLMTest`. Here is the path, which can be easily applied by copying it into clipboard and in IntelliJ IDEA, just go to VCS (top menu) -> Apply patch from clipboard. And commit the changes.

```patch
Index: h2o-algos/src/test/java/hex/glm/GLMTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- h2o-algos/src/test/java/hex/glm/GLMTest.java	(revision 5b5c8f6fb5a7d726dcb0fcdf5002513f18d76024)
+++ h2o-algos/src/test/java/hex/glm/GLMTest.java	(date 1584950978881)
@@ -9,12 +9,11 @@
 import hex.glm.GLMModel.GLMParameters.Solver;
 import hex.glm.GLMModel.GLMWeightsFun;
 import hex.glm.GLMTask.*;
-import org.junit.Assert;
-import org.junit.BeforeClass;
-import org.junit.Ignore;
-import org.junit.Test;
+import org.junit.*;
+import org.junit.rules.ExpectedException;
 import water.*;
 import water.H2O.H2OCountedCompleter;
+import water.exceptions.H2OModelBuilderIllegalArgumentException;
 import water.fvec.*;
 import water.parser.BufferedString;
 import water.parser.ParseDataset;
@@ -27,9 +26,15 @@
 
 import static org.junit.Assert.*;
 
-public class GLMTest  extends TestUtil {
+public class GLMTest extends TestUtil {
 
-  @BeforeClass public static void setup() { stall_till_cloudsize(1); }
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
 
   public static void testScoring(GLMModel m, Frame fr) {
     Scope.enter();
@@ -2180,8 +2185,34 @@
       preds.delete();
       preds2.delete();
       m2.delete();
-    }finally {
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testFractionalBinomialResponse() {
+    try {
+      Scope.enter();
+      final Frame trainingFrame = new TestFrameBuilder().withVecTypes(Vec.T_NUM, Vec.T_NUM)
+              .withColNames("Feature", "Response")
+              .withDataForCol(0, new double[]{1, 2, 3})
+              .withDataForCol(1, new double[]{-0.1, 0.5, 1.1}) // Exceeds min & max values for fractional binom.
+              .build();
+
+      final GLMParameters parameters = new GLMParameters();
+      parameters._train = trainingFrame._key;
+      parameters._family = Family.fractionalbinomial;
+      parameters._response_column = "Response";
+
+      expectedException.expect(H2OModelBuilderIllegalArgumentException.class);
+      expectedException.expectMessage("Details: ERRR on field: response: Response 'Response' must be between 0" +
+              " and 1 for fractional_binomial family. Min: -0.100000, Max: 1.100000");
+      final GLMModel model = new GLM(parameters).trainModel().get();
+      Scope.track_generic(model);
+    } finally {
       Scope.exit();
     }
+
   }
 }
```